format/spec.md

```diff
@@ -1473,7 +1473,10 @@ The following table describes the possible values for the some of the field with
 
 ### Table Metadata and Snapshots
 
-Table metadata is serialized as a JSON object according to the following table. Snapshots are not serialized separately. Instead, they are stored in the table metadata JSON.
+Table metadata is serialized as a JSON object according to the following table. Snapshots are not serialized separately. Instead, they are stored in the table metadata JSON. 
+
+A metadata JSON file must end in `.metadata.json`. A metadata JSON file may be compressed with [GZIP](https://datatracker.ietf.org/doc/html/rfc1952). A GZIP compressed file must end with `.gz.metadata.json`.
```

Comment:
Hi, I have two concerns about this change:

- `.gz.metadata.json` is quite uncommon and can't be read by most existing tools. Would it be better to support `.metadata.json.gz` and treat `.gz.metadata.json` as legacy for backward compatibility?
- `gzip` is becoming increasingly outdated due to its lack of support for modern CPUs. New algorithms like `zstd` are gaining popularity, so should we consider allowing users to use `.metadata.json.zst` as well?

Reply:
@Xuanwo both of these are good points, I don't have historical context on why these conventions where chosen but this is documenting existing behavior, so I think changes should be discussed on the mailing list as a follow-up.

Reply:
Thank you @emkornfield, I have copied my comments to ML.

Reply:
+1 to treating `.metadata.json.gz` as the "default" and `.gz.metadata.json` for backwards compatibility

looking at the broader ecosystem, `.json.gz` is used more often. for example, its in the duckdb documentations
https://duckdb.org/docs/stable/data/json/loading_json.html#parameters

Reply:
Thanks for the feedback, as noted above this is documenting existing behavior.  If we want to change it I think this is a larger discussion. Regardless of what the ecosystem uses to to denote gzip compressed metadata, the fact of the matter is Iceberg chose a different convention.  This is documenting what Java does, and what [Python](https://github.com/apache/iceberg-python/blob/82e025ee73f04e6174bb1b4d853804d609ca5b67/pyiceberg/serializers.py#L35) expects.